Use perlcritic wrapper and plugins from common repo

perlpunk opened the pull request:
Also create a github workflow for it.

Issue: https://progress.opensuse.org/issues/155188

kalikiana commented on .github/workflows/perl-critic.yml:
```diff
@@ -0,0 +1,14 @@
+      image: perldocker/perl-tester
+    steps:
+      - uses: actions/checkout@v4
+      - run: make test-critic
```
```
Policy "Perl::Critic::Policy::HashKeyQuotes" is not installed.
Policy "Perl::Critic::Policy::RedundantStrictWarning" is not installed.
```

I see those in CI 🤔

perlpunk replied:
Thanks, I missed that. Makefile needs to run tools/perlcritic instead. pushed. Now we should see failures and I'm currently trying to fix them

perlpunk commented:
Ok, while trying to fix more perlcritic complaints I found out that the plugins actually need more fixes.
I wish those plugins had been tested on our code base before.

perlpunk commented:
After
* https://github.com/os-autoinst/os-autoinst-common/pull/54
I updated the subrepo.  I added some exceptions now for RedundantStrictWarning, I think that's good enough.

perlpunk commented:
https://app.circleci.com/pipelines/github/perlpunk/openQA/357/workflows/d8ac13a9-ce17-413a-bbc2-8b9779b9e1a2/jobs/2482
```
[13:16:03] t/05-scheduler-full.t .. 1/? 
    #   Failed test 'jobs dispatched to different workers'
    #   at t/05-scheduler-full.t line 119.
    #          got: undef
    #     expected: anything else

    #   Failed test 'each of the two jobs allocated to one of the workers'
    #   at t/05-scheduler-full.t line 120.
    #          got: undef
    #     expected: anything else
    # 0
    # Looks like you failed 2 tests of 4.
[13:16:03] t/05-scheduler-full.t .. 2/? 
#   Failed test 'Scheduler worker job allocation'
#   at t/05-scheduler-full.t line 128.
...
```
This seems unrelated. I'll retrigger

perlpunk commented:
Created https://progress.opensuse.org/issues/165198 for the unrelated failing test

perlpunk commented:
So there seems to be a problem with circleci - I retriggered the failing test and it passed now, but the links in the github checks still link to the older run.
Also the codecov upload failed with the typical token error.
Let's see what happens when I rebase to trigger a completely new run
